pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -219,9 +218,19 @@ def fetch_mirror(remote):
             match = re.match(url_pattern, mirror)
             if not match:
                 continue
-            repodata_exists = get_repomd_file(remote, match.group(2))
-            if match and repodata_exists:
-                return match.group(2)
+
+            mirror_url = match.group(2)
+            try:
+                repodata_exists = get_repomd_file(remote, mirror_url, raise_err=True)
+                if repodata_exists:
```

Comment:
Can this ever be None now? Either you found something, or an exception was raised, yeah?